shadows/framework/src/main/java/org/robolectric/shadows/ShadowBitmapFactory.java

```diff
@@ -231,6 +232,14 @@ public static Bitmap create(final String name, final BitmapFactory.Options optio
       options.outWidth = p.x;
       options.outHeight = p.y;
     }
+
+    if (options != null && RuntimeEnvironment.getApiLevel() >= Build.VERSION_CODES.KITKAT) {
+      ReflectionHelpers.callStaticMethod(
+          BitmapFactory.class,
+          "setDensityFromOptions",
+          ClassParameter.from(Bitmap.class, bitmap),
+          ClassParameter.from(BitmapFactory.Options.class, options));
+    }
```

Comment:
I think this should do something for SDK < 19 as well:

```java
    if (RuntimeEnvironment.getApiLevel() >= Build.VERSION_CODES.KITKAT) {
      ReflectionHelpers.callStaticMethod(
          BitmapFactory.class,
          "setDensityFromOptions",
          ClassParameter.from(Bitmap.class, bitmap),
          ClassParameter.from(BitmapFactory.Options.class, options));
    } else if (options != null && options.inDensity != 0) {
      bitmap.setDensity(options.inDensity);
    }
```

Reply:
I copy `setDensityFromOptions` from `BitmapFactory`, it makes the logic more complete and makes the code more readable

Reply:
We have a pretty strong preference to call the real Android code wherever possible. On Android < 19, the function is named [finishDecode](https://cs.android.com/android/platform/superproject/+/android-4.3.1_r1:frameworks/base/graphics/java/android/graphics/BitmapFactory.java;l=556), but most of that does not seem too relevant except the `bm.setDensity(density);`.

Reply:
Get, I will call the real Android code